src/python/pants/backend/experimental/python/lockfile_metadata.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pants.backend.experimental.python.lockfile import PythonLockfileRequest
+
```

Comment:
Hm comment isn't showing up anymore, but bump on avoiding the circular import by taking the hex_digest directly in the arg for `lockfile_content_with_header`

Reply:
Beat you to it :)